packages/plugins/RedPacket/src/SiteAdaptor/RedPacket/useRedPacketCover.ts

```diff
@@ -0,0 +1,39 @@
+import type { RedPacketJSONPayload } from '@masknet/web3-providers/types'
+import { useQuery, type UseQueryResult } from '@tanstack/react-query'
+import type { useAvailability } from '../hooks/useAvailability.js'
+import { FireflyRedPacket } from '@masknet/web3-providers'
+import { useMemo } from 'react'
+import { isValidAddress, isValidDomain } from '@masknet/web3-shared-evm'
+import { minus, toFixed } from '@masknet/web3-shared-base'
+
+type Availability = ReturnType<typeof useAvailability>['data']
+type T = UseQueryResult
+export function useRedPacketCover(payload: RedPacketJSONPayload, availability: Availability) {
+    const { data: theme } = useQuery({
+        queryKey: ['red-packet', 'theme-by-id', payload.rpid],
+        queryFn: async () => FireflyRedPacket.getThemeById(payload.rpid),
```

Comment:
Use `getCoverUrlByRpid` and you can compose `coverUrl` in that method.

Reply:
Never mind the redundant API requests; utilize `fetchJSON` to cache the response.